File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int[] count = {0, 0};
        int n = students.length, i;

        for(int val : students) {
            count[val]++;
        }

        for(i = 0; i<n && count[sandwiches[i]] > 0; i++) {
            count[sandwiches[i]]--;
        }

        return n - i;
    }
}

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int count[] = {0, 0};
        int n = students.size(), i;

        for(int choice : students) {
            count[choice]++;
        }

        for(i = 0; i<n && count[sandwiches[i]] > 0; i++) {
            count[sandwiches[i]]--;
        }

        return n - i;
    }
};